.travis.yml

```diff
@@ -0,0 +1,103 @@
+language: c
+dist: trusty
+sudo: false
+group: beta
+
+# To cache doc-building dependencies.
+cache: pip
+
+branches:
+  only:
+    - master
+    - /^\d\.\d$/
+
+matrix:
+  fast_finish: true
+  allow_failures:
+    - env: OPTIONAL=true
+  include:
+    - os: linux
+      language: c
+      compiler: clang
+      # gcc also works, but to keep the # of concurrent builds down, we use one C
+      # compiler here and the other to run the coverage build. Clang is preferred
+      # in this instance for its better error messages.
+      env: TESTING=cpython
+    - os: osx
+      language: c
+      compiler: clang
+      # Testing under macOS is optional until testing stability has been demonstrated.
+      env: OPTIONAL=true
+      before_install:
+        - brew install openssl xz
+        - export CPPFLAGS="-I$(brew --prefix openssl)/include"
+        - export LDFLAGS="-L$(brew --prefix openssl)/lib"
+    - os: linux
+      language: python
+      # Build the docs against a stable version of Python so code bugs don't hold up doc-related PRs.
+      python: 3.6
+      env: TESTING=docs
+      before_script:
+        - cd Doc
+        # Sphinx is pinned so that new versions that introduce new warnings won't suddenly cause build failures.
+        # (Updating the version is fine as long as no warnings are raised by doing so.)
+        - python -m pip install sphinx~=1.6.1
+      script:
+        - make check suspicious html SPHINXOPTS="-q -W -j4"
+    - os: linux
+      language: c
+      compiler: gcc
+      env: OPTIONAL=true
+      before_script:
+        - |
+            if ! git diff --name-only $TRAVIS_COMMIT_RANGE | grep -qvE '(\.rst$)|(^Doc)|(^Misc)'
+            then
+              echo "Only docs were updated, stopping build process."
+              exit
+            fi
+            ./configure
+            make -s -j4
+            # Need a venv that can parse covered code.
+            ./python -m venv venv
+            ./venv/bin/python -m pip install -U coverage
+      script:
+        # Skip tests that re-run the entire test suite.
+        - ./venv/bin/python -m coverage run --pylib -m test -uall,-cpu,-tzdata -x test_multiprocessing_fork -x test_multiprocessing_forkserver -x test_multiprocessing_spawn
+      after_script:  # Probably should be after_success once test suite updated to run under coverage.py.
+        # Make the `coverage` command available to Codecov w/ a version of Python that can parse all source files.
+        - source ./venv/bin/activate
+        - bash <(curl -s https://codecov.io/bash)
```

Comment:
With the codecov config missing, you can just remove this entire build.  If you still want GCC coverage, you could change the main Linux build to GCC, since clang is covered by the OSX builder.